### ISSUE TITLE
SNOW-982793 SNOW-993512: SNYK-JS-INFLIGHT-6095116: High Vulnerability via glob -> inflight

### ISSUE THREAD

**Issue by junaid-ali** [snowflakedb/snowflake-connector-nodejs#720]
https://github.com/snowflakedb/snowflake-connector-nodejs/blob/master/package.json#L25

https://security.snyk.io/vuln/SNYK-JS-INFLIGHT-6095116

fix is to upgrade glob version to v9+ which no longer depends on the vulnerable package: https://github.com/isaacs/inflight/issues/5#issuecomment-1835585106

**Comment by sfc-gh-dszmolka** [comment]
thank you for raising this issue - already in progress under https://github.com/snowflakedb/snowflake-connector-nodejs/pull/714

**Comment by sfc-gh-dszmolka** [comment]
PR is merged and will be part of the next release, expected by latest next week

**Comment by sfc-gh-dszmolka** [comment]
reopening because it looks like its not that easy to get rid of `inflight`. still with us, due to a transitive dependency .

```
root@ef641033e6e3:/node# npm why glob
glob@9.3.5 # <-- this is the fixed version
node_modules/glob
  glob@"^9.0.0" from snowflake-sdk@1.9.2
  node_modules/snowflake-sdk
    snowflake-sdk@"^1.9.2" from the root project

glob@7.2.3 # <-- sadly this one still has `inflight`
node_modules/rimraf/node_modules/glob
  glob@"^7.1.3" from rimraf@3.0.2
  node_modules/rimraf
    rimraf@"^3.0.0" from tmp@0.2.1
    node_modules/tmp
      tmp@"^0.2.1" from snowflake-sdk@1.9.2
      node_modules/snowflake-sdk
        snowflake-sdk@"^1.9.2" from the root project

root@ef641033e6e3:/node# npm why inflight
inflight@1.0.6
node_modules/inflight
  inflight@"^1.0.4" from glob@7.2.3
  node_modules/rimraf/node_modules/glob
    glob@"^7.1.3" from rimraf@3.0.2
    node_modules/rimraf
      rimraf@"^3.0.0" from tmp@0.2.1
      node_modules/tmp
        tmp@"^0.2.1" from snowflake-sdk@1.9.2
        node_modules/snowflake-sdk
          snowflake-sdk@"^1.9.2" from the root project
```

**Comment by junaid-ali** [comment]
seems like rimraf should be upgraded to atleast [v4.2.0](https://github.com/isaacs/rimraf/blob/v4.2.0/package-lock.json#L12)?

**Comment by junaid-ali** [comment]
https://github.com/isaacs/rimraf/blob/v4.2.0/CHANGELOG.md#v42

**Comment by sfc-gh-dszmolka** [comment]
indeed there are multiple options (override dependency, replace `tmp`, etc.), the team will look into how to proceed best and i'll keep this thread posted

**Comment by sfc-gh-dszmolka** [comment]
fix in progress under https://github.com/snowflakedb/snowflake-connector-nodejs/pull/742 (removing `tmp` -> `glob` v7 -> `inflight`)

**Comment by sfc-gh-dszmolka** [comment]
PR merged and now `inflight` is gone with removing dependency on `tmp`

```
# npm i snowflake-sdk@1.9.2
..
# npm why inflight
inflight@1.0.6
node_modules/inflight
  inflight@"^1.0.4" from glob@7.2.3
  node_modules/rimraf/node_modules/glob
    glob@"^7.1.3" from rimraf@3.0.2
    node_modules/rimraf
      rimraf@"^3.0.0" from tmp@0.2.1
      node_modules/tmp
        tmp@"^0.2.1" from snowflake-sdk@1.9.2
        node_modules/snowflake-sdk
          snowflake-sdk@"^1.9.2" from the root project 

# install latest from main
# npm i https://github.com/snowflakedb/snowflake-connector-nodejs.git
..
# npm why inflight
npm ERR! No dependencies found matching inflight
```

will be part of the next (January) release, expected within 2 weeks

**Comment by sfc-gh-dszmolka** [comment]
released with snowflake-sdk 1.9.3
